backend/Search.hs

```diff
@@ -209,24 +218,26 @@ configOptP = do
 
 main :: IO ()
 main = do
+  register ghcMetrics
   config@Config{configBindTarget} <-
     Opt.execParser $
       Opt.info (configOptP <**> Opt.helper)
         (Opt.fullDesc <> Opt.header "Hackage Search")
-  runServer configBindTarget (serve hackageSearchAPI (hackageSearchServer config))
+  runServer configBindTarget . serve metricsApiProxy . withMetrics hackageSearchAPI $
+    hackageSearchServer config
 
 data BindTarget
   = BindOnPort Int
   | BindOnUnixSocket FilePath
 
 runServer :: BindTarget -> Application -> IO ()
-runServer (BindOnPort port) = run port
-runServer (BindOnUnixSocket path) = \app ->
-  Network.Socket.withSocketsDo $
-  bracket
-    (Data.Streaming.Network.bindPath path)
-    Network.Socket.close
-    (\socket -> runSettingsSocket defaultSettings socket app)
+runServer target app = register (meters metricsApiProxy) >>= \mtrs ->
+  case target of
+    BindOnPort port       -> run port $ monitorServant mtrs app
+    BindOnUnixSocket path -> withSocketsDo $
+      bracket (bindPath path) close $ \socket -> do
+        mtrs <- register $ meters metricsApiProxy
```

Comment:
Why create `mtrs` a second time here?

Reply:
Missed that upon refactoring, thank you. Pushed a fix!